cheroot/connections.py

```diff
@@ -107,88 +118,61 @@ def get_conn(self, server_socket):
         Any connection returned by this method will need to be `put`
         back if it should be examined again for another request.
 
-        Args:
-            server_socket (socket.socket): Socket to listen to for new
-            connections.
         Returns:
             cheroot.server.HTTPConnection instance, or None.
 
         """
         # Grab file descriptors from sockets, but stop if we find a
         # connection which is already marked as ready.
-        socket_dict = {}
-        for conn in self.connections:
-            if conn.closeable or conn.ready_with_data:
-                break
-            socket_dict[conn.socket.fileno()] = conn
-        else:
-            # No ready connection.
-            conn = None
 
-        # We have a connection ready for use.
-        if conn:
-            self.connections.remove(conn)
-            return conn
+        try:
+            return self._readable_conns.popleft()
+        except IndexError:
+            pass
```

Comment:
I think it's better semantically to use suppress:
```suggestion
        with suppress(IndexError):
            return self._readable_conns.popleft()
```
_(add `from ._compat import suppress` at the top)_